core/server/update-check.js

```diff
@@ -142,78 +164,123 @@ function updateCheckRequest() {
                 reject(error);
             });
 
-            req.write(reqData);
+            req.write(snapshot);
             req.end();
         });
     });
 }
 
-// ## Update Check Response
-// Handles the response from the update check
-// Does two things with the information received:
-// 1. Updates the time we can next make a check
-// 2. Checks if the version in the response is new, and updates the notification setting
+/**
+ * Update Check Response. Handles the response from UpdateCheck service. On response:
+ * 1. Updates the time for next checkin
+ * 2. Checks if the version in the response is new, and updates the notification setting
+ * @param {Object} response body after calling UpdateCheck service
+ * @returns {Promise}
+ */
 function updateCheckResponse(response) {
-    var ops = [];
-
-    ops.push(
-        api.settings.edit(
-            {settings: [{key: 'nextUpdateCheck', value: response.next_check}]},
-            internal
-        ),
-        api.settings.edit(
-            {settings: [{key: 'displayUpdateNotification', value: response.version}]},
-            internal
-        )
-    );
-
-    return Promise.all(ops);
+    return Promise.all([
+        api.settings.edit({settings: [{key: 'nextUpdateCheck', value: response.next_check}]}, internal),
+        api.settings.edit({settings: [{key: 'displayUpdateNotification', value: response.version || response.release.name}]}, internal)
```

Comment:
`response.version || response.release.name`

Don't think this change is required anymore.